views/013_updated_at_column_trigger.sql

```diff
@@ -0,0 +1,28 @@
+CREATE OR REPLACE FUNCTION update_updated_at_column()
+RETURNS TRIGGER AS $$
+BEGIN
+  NEW.updated_at = NOW();
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Iterate over all tables (excluding views) in the current schema and 
+-- create a trigger on each table that has an "updated_at" column
+DO $$ 
+DECLARE 
+  tbl_name TEXT;
+BEGIN
+  FOR tbl_name IN
+    SELECT table_name
+    FROM information_schema.columns
+    WHERE 
+      table_schema = current_schema() 
+      AND column_name = 'updated_at'
+      AND table_name NOT IN (SELECT table_name FROM information_schema.views WHERE table_schema = current_schema()) 
+  LOOP
+    EXECUTE format('CREATE OR REPLACE TRIGGER %I_update_updated_at
```

Comment:
Does this not loop? Do we not need fire if a) any of the fields have changed and b) updated_at is not in the change list ?

Reply:
No, this does not run into an infinite loop because the `update_updated_at_column` trigger function simply modifies the row before update happens. The function itself doesn't run any SQL statements to perform the update.

> any of the fields have changed

I've made this modification so the `updated_at` column is set to `NOW()` only if there's some modification to the row that's being updated.